vmdb/app/models/orchestration_stack_amazon.rb

```diff
@@ -1,13 +1,38 @@
 class OrchestrationStackAmazon < OrchestrationStack
+  def self.raw_create_stack(orchestration_manager, stack_name, template, options = {})
+    orchestration_manager.with_provider_connection(:service => "Orchestration") do |service|
+      service.stacks.create(stack_name, template.content, options).stack_id
+    end
+  rescue => err
+    _log.error "MIQ(#{self.class.name}##{__method__}) stack=[#{stack_name}], error: #{err}"
+    raise MiqException::MiqOrchestrationProvisionError, err.to_s, err.backtrace
+  end
+
   def raw_update_stack(options)
-    ext_management_system.with_provider_connection(:service => "CloudFormation") do |connection|
-      connection.stacks[name].update(options)
+    ext_management_system.with_provider_connection(:service => "CloudFormation") do |service|
+      service.stacks[name].update(options)
     end
+  rescue => err
+    _log.error "MIQ(#{self.class.name}##{__method__}) stack=[#{name}], error: #{err}"
+    raise MiqException::MiqOrchestrationUpdateError, err.to_s, err.backtrace
   end
 
   def raw_delete_stack
-    ext_management_system.with_provider_connection(:service => "CloudFormation") do |connection|
-      connection.stacks[name].delete
+    ext_management_system.with_provider_connection(:service => "CloudFormation") do |service|
```

Comment:
Raw methods are provider API dependent. For the two subclasses we currently have, they seem to have some similarity and suggest we move the implementation to parent. I would not assume other provider APIs that we might add in the future have the same pattern.